### PR TITLE
Tile highlighting hexxle 126

### PR DESCRIPTION
**Pull request by RicardoMonteiroSimoes**

### CODE REVIEW

**Comment by DennisBannerman** [comment]
Testing it right now. Things I noticed (will edit, if I find more):

* If you open the pausemenu, you can still click behind it, tiles are also still highlighted. This functionality was disabled in a previous commit. Maybe change it back/reimplement unless that is intended behaviour.

**Comment by RicardoMonteiroSimoes** [comment]
> Testing it right now. Things I noticed (will edit, if I find more):
> 
> * If you open the pausemenu, you can still click behind it, tiles are also still highlighted. This functionality was disabled in a previous commit. Maybe change it back/reimplement unless that is intended behaviour.
> * I seem to have Recovery GameObjects in my main scene - is that only the case on my machine or do you have those too?

I deleted those objects, I assume some relics I must've overlooked after the rebase.

I fixed those bugs and added tests to cover it, should be okay now

File: unity/Assets/Scripts/Unity/Input/MouseEventsHandler.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            oldTile = null;
+            currentCollisionTile = null;
+        }
+
+        // Start is called before the first frame update
+        void Start()
+        {
+            var inputManager = new InputManager();
+            inputManager.TilePlacement.MouseClick.Enable();
+            inputManager.TilePlacement.MouseClick.performed += context => MouseClickAction();
```
Comment (DennisBannerman) [inline unity/Assets/Scripts/Unity/Input/MouseEventsHandler.cs:26]: I'd move this to OnEnable() and also make sure, that OnDisable() is implemented and strips `performed` of its Eventhandler method. See EscapeEventHandler class.

File: unity/Assets/Scripts/Unity/Input/MouseEventsHandler.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        }
+
+        public void ResumeGame()
+        {
+            isGamePaused = false;
+            if (currentCollisionTile != null)
+            {
+                currentCollisionTile.GetComponent<UnityTileHighlighter>().TurnOn();
+            }
+        }
```
Comment (DennisBannerman) [inline unity/Assets/Scripts/Unity/Input/MouseEventsHandler.cs:74]: Does UnityTileHighlighter's `enabled` property not already provide the TurnOn/TurnOff logic? Setting enabled to true/false should fire `OnDisable` and `OnEnable`, which you chould also override if you'd need some custom behaviour.

File: unity/Assets/Scripts/Unity/UnityStack.cs [hunk 3]
```diff
@@ -56,7 +56,7 @@ private void DisplayStack()
             {
                 ITile tile = firstTen.ElementAt(i);
                 Texture texture = GetTextureForTileType(tile);
-                GameObject newTile = Instantiate(Template, Content.transform);
+                GameObject newTile = Instantiate(Template);
```
Comment (DennisBannerman) [inline unity/Assets/Scripts/Unity/UnityStack.cs:59]: What does this change? Just curious.

File: unity/Assets/Scripts/Unity/UnityTileHighlighter.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        void Start()
+        {
+            isHighlighted = false;
+        }
+
+        // Update is called once per frame
+        void Update()
+        {
+
+        }
```
Comment (DennisBannerman) [inline unity/Assets/Scripts/Unity/UnityTileHighlighter.cs:26]: Looks like unused code.

File: unity/Assets/Scripts/Unity/UnityTileHighlighter.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        {
+            isHighlighted = true;
+            setToBlackOnce = false;
+        }
+
+        public void TurnOff()
+        {
+            isHighlighted = false;
+            setToBlackOnce = true;
+        }
```
Comment (DennisBannerman) [inline unity/Assets/Scripts/Unity/UnityTileHighlighter.cs:55]: As noted in another class - is this required if we can just enable or disable the component?

**Review by DennisBannerman — APPROVED** [review]
Looks good, got some minor changes. They don't change the basic logic though so feel free to merge at your own discretion.

File: unity/Assets/Scripts/Unity/UnityStack.cs (hunk 3 above)
Comment (RicardoMonteiroSimoes) [reply unity/Assets/Scripts/Unity/UnityStack.cs:59]: I had a warning telling me there are too many transforms going on, removing this parameter fixed it.

File: unity/Assets/Scripts/Unity/Input/MouseEventsHandler.cs (hunk 2 above)
Comment (RicardoMonteiroSimoes) [reply unity/Assets/Scripts/Unity/Input/MouseEventsHandler.cs:74]: no idea, didn't think of that before. I'll try it out

**Comment by RicardoMonteiroSimoes** [comment]
Refactored the code to use `enabled` to turn on and off, works exactly the same but with much cleaner code

**Review by DennisBannerman — APPROVED** [review]

**Pull request merged**